src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -377,6 +378,37 @@ void removeReference() {
         }
     }
 
+    /**
+     * Keeps track of the last network activity in order to tell if the connection has been idle. This is done to avoid
+     * checking the socket for a disconnect every command execution, which has a high impact on perf.
+     */
+    class IdleNetworkTracker {
+        private Instant lastNetworkActivity = Instant.now();
+        private int maxIdleMillis = 15000;
```

Comment:
> why 15000?

An “idle” connection will only ever gets its socket disconnected by a keepalive packet after a connection has been severed. KeepAlive packets are only sent on idle sockets. I don’t see keepalives ever being configured for more frequent than 15s. Default setting by the driver (on platforms that have Java support for setting it) and the recommended setting is 30s (and OS default for those that don't set it is 2 hrs). Since checking for a dead connection on a non-idle socket would have a high perf impact, we don’t want to do it too often.